parquet-hadoop/src/main/java/org/apache/parquet/filter2/compat/RowGroupFilter.java

```diff
@@ -38,21 +39,47 @@
  * no filtering will be performed.
  */
 public class RowGroupFilter implements Visitor<List<BlockMetaData>> {
+  private static final int STATISTICS_FIXED_VERSION = 161; // 1.6.1
   private final List<BlockMetaData> blocks;
   private final MessageType schema;
+  private final Boolean skipStatisticsChecks;
 
-  public static List<BlockMetaData> filterRowGroups(Filter filter, List<BlockMetaData> blocks, MessageType schema) {
+  public static List<BlockMetaData> filterRowGroups(Filter filter, ParquetMetadata footer,
+                                                    MessageType schema) {
     checkNotNull(filter, "filter");
-    return filter.accept(new RowGroupFilter(blocks, schema));
+    return filter.accept(new RowGroupFilter(footer, schema));
   }
 
-  private RowGroupFilter(List<BlockMetaData> blocks, MessageType schema) {
-    this.blocks = checkNotNull(blocks, "blocks");
+  private RowGroupFilter(ParquetMetadata footer, MessageType schema) {
+    String createdBy = checkNotNull(footer.getFileMetaData().getCreatedBy(), "createdBy");
+    this.skipStatisticsChecks = shouldIgnoreStatistics(createdBy);
+    this.blocks = checkNotNull(footer.getBlocks(), "blocks");
     this.schema = checkNotNull(schema, "schema");
   }
 
+  private static boolean shouldIgnoreStatistics(String createdBy) {
+    final String[] versionTokens = createdBy.split(" ");
+
+    if (versionTokens.length < 3) {
+      return true;
+    }
+
+    final String app = versionTokens[0];
+    final String version = versionTokens[2].substring(0, 5).replaceAll("\\.", "");
+
+    if (app.equalsIgnoreCase("parquet-mr")) {
+      return Integer.parseInt(version) < STATISTICS_FIXED_VERSION;
+    }
+
+    return true;
+  }
+
   @Override
   public List<BlockMetaData> visit(FilterCompat.FilterPredicateCompat filterPredicateCompat) {
+    if (skipStatisticsChecks) {
```

Comment:
It looks like this would always skip statistics, but there are a lot of cases where statistics are still valid. I think they only need to be skipped when the underlying column type is binary, fixed, and int96 because the other objects are immutable (string) or primitives. In that case, the checks here should only ignore statistics for the potentially corrupted columns.